cmd/ls/ls_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var tempDir string
+
+func TestListCurrentDir(t *testing.T) {
+	buf := new(bytes.Buffer)
+	files, _ := ioutil.ReadDir(tempDir)
+	printFileList(files, buf)
+	output := fmt.Sprint(buf)
+	expected := "-rw-rw-r--   24 f1.txt\n-rw-rw-r--   24 f2.pdf\n-rw-rw-r--   24 f3\n-rw-rw-r--   24 file with space\n"
+	if output != expected {
+		t.Errorf("result not expected [%v]", output)
+	}
+}
+
+func TestListCurrentDirAsAList(t *testing.T) {
+	buf := new(bytes.Buffer)
+	files, _ := ioutil.ReadDir(tempDir)
+	printFileNames(files, buf)
+	output := fmt.Sprint(buf)
+	expected := "f1.txt  f2.pdf  f3  'file with space'  \n"
+	if output != expected {
+		t.Errorf("result not expected [%v]", output)
+	}
+}
+
+func createTempFiles(dir string) {
+	content := []byte("temporary file's content")
+	files := []string{"file with space", "f1.txt", "f2.pdf", "f3"}
+
+	for _, f := range files {
+		fileName := filepath.Join(dir, f)
+		if err := ioutil.WriteFile(fileName, content, 0666); err != nil {
+			os.Exit(1)
+		}
+	}
+}
+
+func setup() {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		os.Exit(1)
+	}
+
+	tempDir = dir
+	createTempFiles(tempDir)
+}
+
+func shutdown() {
+	defer os.RemoveAll(tempDir)
+}
+
+func TestMain(m *testing.M) {
```

Comment:
What do you think of move the TestMain to top of the file? It helps understand the setups/teardowns of overall tests.